File: OJ/leet389/leet389.py
# leet389

from typing import List

from collections import Counter, defaultdict
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        S = defaultdict(int, Counter(s))
        T = Counter(t)
        for ti, cnt in T.items():
            if S[ti] != cnt:
                return ti 

print(Solution().findTheDifference(s = "abcd", t = "abcde"))
print(Solution().findTheDifference(s = "a", t = "aa"))

File: OJ/leet389/leet389.cpp
/*
leet389


*/


#include <iostream>
#include <vector>
#include <queue>
#include <stack>
#include <string>
#include <unordered_set>
#include <unordered_map>
#include <algorithm>
#include <tuple>
#include <cassert>
#include <utility>
using namespace std;

class Solution {
public:
    char findTheDifference(string s, string t) {
        // 异或运算 Time (n+m) Space O(1)
        char ans = 0;
        for (auto ch: s) ans ^= ch;
        for (auto ch: t) ans ^= ch;
        return ans;
    }
};


int main(int argc, char const *argv[])
{
    Solution sol;   
    cout << sol.findTheDifference("abcd", "abcde") << endl;
    cout << sol.findTheDifference("a", "aa") << endl;
    return 0;
}
